corehq/apps/sms/tasks.py

```diff
@@ -129,6 +129,32 @@ def message_is_stale(msg, utcnow):
         return True
 
 
+def connection_slot_key_base(backend):
+    return 'backend-%s-connection-slot-' % backend.couch_id
+
+
+def reserve_connection_slot(backend, max_simultaneous_connections):
+    with CriticalSection(['reserve-connection-slot-for-%' % backend.couch_id]):
+        client = get_redis_client()
+        key_base = connection_slot_key_base(backend)
+        slot_keys = client.keys(key_base + '*')
+        slot_numbers = [int(slot_key.replace(key_base, '')) for slot_key in slot_keys]
+
+        for slot_number in range(1, max_simultaneous_connections + 1):
+            if slot_number not in slot_numbers:
+                key = key_base + slot_number
+                client.set(key, 1)
+                client.expire(key, 60)
```

Comment:
Is this 60 seconds?  Was it just chosen arbitrarily?

Reply:
yeah 60 seconds. it should be large enough to cover an http request to a gateway, but small enough that waiting for it to timeout if a task drops half-way through doesn't take too long, so 1 minute seemed appropriate.